File: template/android/app/src/main/kotlin/com/sablue/sotwo/security_keypad/SecurityKeypadViewFactory.kt
package com.sablue.sotwo.security_keypad

import android.content.Context
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory
import java.lang.Exception

class SecurityKeypadViewFactory : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        val creationParams = args as Map<String?, Any?>
        val maxInputLength = when(creationParams["viewType"] as String) {
            // TODO : Flutter로부터 전달받도록 수정
            "password" -> 4
            "residentNumber" -> 13
            else -> throw Exception("Wrong viewType")
        }
        return SecurityKeypadView(context, maxInputLength)
    }
}

File: template/ios/Runner/Core/SecurityKeypad/SecurityKeypadViewFactory.swift
//
//  SecurityKeypadViewFactory.swift
//  Runner
//
//  Created by SAB-150 on 1/31/24.
//

import Flutter

class SecurityKeypadViewFactory: NSObject, FlutterPlatformViewFactory {
    private var messenger: FlutterBinaryMessenger
	private let flutterApi: SecurityKeypadFlutterApi

    init(messenger: FlutterBinaryMessenger) {
        self.messenger = messenger
		self.flutterApi = SecurityKeypadFlutterApi(binaryMessenger: messenger)
        super.init()
    }

    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
		var maxInputLength = 0
		if let creationParams = args as? [String : Any?] {
			// TODO : Flutter로부터 전달받도록 수정
			maxInputLength = switch creationParams["viewType"] as? String {
			case "password": 4
			case "residentNumber": 13
			default: 0
			}
		}
        return SecurityKeypadView(
            frame: frame,
            viewIdentifier: viewId,
            arguments: args,
            binaryMessenger: messenger,
			maxInputLength: maxInputLength,
			flutterApi: flutterApi
        )
    }

    /// Implementing this method is only necessary when the `arguments` in `createWithFrame` is not `nil`.
    public func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
          return FlutterStandardMessageCodec.sharedInstance()
    }
}
